### PR TITLE
[POAE7-2664] add context buffer

### PR DESCRIPTION
**Pull request by YantingTao1315**
### What changes were proposed in this pull request?
+ add buffer related function in codegen context

### Why are the changes needed?
+ buffer registration used by agg and etc.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
UTs

### Which label does this PR belong to?
INFRA

### CODE REVIEW

File: cider/exec/nextgen/context/Buffer.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef NEXTGEN_CONTEXT_BUFFER_H
+#define NEXTGEN_CONTEXT_BUFFER_H
+
+#include "include/cider/batch/CiderBatchUtils.h"
```
Comment (xieqi) [inline cider/exec/nextgen/context/Buffer.h:24]: don't need include

File: cider/exec/nextgen/context/Buffer.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef NEXTGEN_CONTEXT_BUFFER_H
+#define NEXTGEN_CONTEXT_BUFFER_H
+
+#include "cider/CiderAllocator.h"
+
+namespace cider::exec::nextgen::context {
+class Buffer {
```
[the remark this answers is not in the document]
Comment (YantingTao1315) [reply cider/exec/nextgen/context/Buffer.h:27]: Non-groupby Agg output

File: cider/exec/nextgen/context/CodegenContext.cpp [hunk 3]
```diff
@@ -54,6 +54,36 @@ JITValuePointer CodegenContext::registerBatch(const SQLTypeInfo& type,
   return ret;
 }
 
+JITValuePointer CodegenContext::registerBuffer(const int32_t capacity,
+                                               const std::string& name,
+                                               bool under_level_output) {
```
Comment (bigPYJ1151) [inline cider/exec/nextgen/context/CodegenContext.cpp:59]: Maybe ```output_raw_buffer``` is better.

**Review by Ma-Jian1 — DISMISSED** [review]
LGTM

**Review by bigPYJ1151 — APPROVED** [review]

**Review by Ma-Jian1 — APPROVED** [review]

**Pull request merged**